### PR TITLE
Add SetMetricType and set metric only for shard leader

### PR DESCRIPTION
**Pull request by bigsheeper**
/kind improvement

The primary objectives of this PR are:
1. To save DescribeIndex RPC each time querycoord wraps LoadSegmentTask;
2. Let the metric type logic remain consistent with the 2.2.x version;

### CODE REVIEW

**Review by yah01 — APPROVED** [review]
/lgtm

File: internal/querycoordv2/task/executor.go [hunk 1]
```diff
@@ -254,21 +254,9 @@ func (ex *Executor) loadSegment(task *SegmentTask, step int) error {
...
-	metricType, err := getMetricType(indexInfo, schema)
-	if err != nil {
-		log.Warn("failed to get metric type", zap.Error(err))
-		return err
-	}
-
 	loadMeta := packLoadMeta(
 		ex.meta.GetLoadType(task.CollectionID()),
-		metricType,
+		"",
```
[the remark this answers is not in the document]
Comment (bigsheeper) [reply internal/querycoordv2/task/executor.go:259]: I'll fix it later

**Comment by bigsheeper** [comment]
related：https://github.com/milvus-io/milvus/issues/25205

**Pull request merged**